internal/tofu/evaluate.go

```diff
@@ -958,7 +960,16 @@ func (d *evaluationStateData) GetResource(ctx context.Context, addr addrs.Resour
 		} else {
 			ret = cty.EmptyObjectVal
 		}
-
+	case config.Enabled != nil:
+		val, ok := instances[addrs.NoKey]
+		if ok {
+			ret = val
+		} else {
+			// When we're using enabled, it's okay to have no
+			// instance, and the entire object is null because
+			// this needs to be propagated to the caller.
+			ret = cty.NullVal(cty.DynamicPseudoType)
```

Comment:
Most of the other return paths from this function seem to use `ty` -- the `cty.Type` implied by the schema of the resource type -- as their return type when returning unknown values, so probably we could use `cty.NullVal(ty)` here to give more type information in case that helps catch another problem elsewhere in the configuration.